### ISSUE TITLE
mailcap: handle reddit gallery like imgur gallery

### ISSUE THREAD

**Issue by henkmet** [tildeclub/ttrv#35]
Pretty much as the title says. It would be great if a reddit gallery could be opened in feh when using mailcap. Maybe it's possible by just adding a correct line to the mailcap file, but I'm not sure how to get the mimetype and how to extract the urls.
Thanks

**Comment by pianoslum** [comment]
Do you have an example link? I also noticed that some reddit links get opened in the browser instead of feh...

**Comment by henkmet** [comment]
For example: https://www.reddit.com/gallery/l8baz8

**Comment by pianoslum** [comment]
The root problem seems to be that feh is not able to load the gallery, for me it displays feh WARNING: https://www.reddit.com/gallery/l8baz8 - No Imlib2 loader for that file format
Or did you find a way?

**Comment by henkmet** [comment]
No, I was asking for a way, actually. If it's not possible then it's too bad but no big deal of course.

**Comment by pianoslum** [comment]
I opened a feature request for feh https://github.com/derf/feh/issues/591
Adding it to the mailcap afterwards should be trivial

**Comment by gbpi** [comment]
It is a ttrv problem, not feh problem. For Imgur albums ttrv uses the Imgur API to get a list of the image URLs, then feeds those to feh or whatever viewer you choose. They have no reason to support albums from some random host, they only load direct image URLs.

**Comment by gbpi** [comment]
Taking his example, go to the main thread URL and add .json.

https://www.reddit.com/r/minipainting/comments/l8baz8/skink_oracle_on_troglodon/.json

The image URLs needed to feed into the viewer are there.

Doesn't seem terribly hard to implement for someone more capable than me.

**Comment by gbpi** [comment]
I've done some research, I'm not a capable programmer, but someone should be able to implement this.

Take his example of https://www.reddit.com/gallery/l8baz8, the thread URL is https://www.reddit.com/r/minipainting/comments/l8baz8/skink_oracle_on_troglodon/.

Add .json to the end of that thread URL and look for gallery_data.

https://www.reddit.com/r/minipainting/comments/l8baz8/skink_oracle_on_troglodon/.json

`"gallery_data": {"items": [{"media_id": "4lmf94hrpde61", "id": 25109829}, {"media_id": "bj36xysrpde61", "id": 25109830}, {"media_id": "obz0mx2spde61", "id": 25109831}, {"media_id": "p48vnccspde61", "id": 25109832}]}`

https://i.redd.it/4lmf94hrpde61.jpg
https://i.redd.it/bj36xysrpde61.jpg
https://i.redd.it/obz0mx2spde61.jpg
https://i.redd.it/p48vnccspde61.jpg

There are your URLs to feed to feh or whatever image viewer has the same capability. I know pqiv works roughly the same as feh.
